backend/src/main/java/de/neuefische/backend/service/RecipeService.java

```diff
@@ -29,4 +33,22 @@ public List<RecipeNormalized> getAllRecipes() {
         )).toList();
     }
 
+    public RecipeNormalized getRecipeById(String id) {
+        return repo
+                .findById(id).stream().map(recipe -> new RecipeNormalized(
+                        recipe.getId(),
+                        recipe.getName(),
+                        recipe.getDescription(),
+                        recipe.getInstructions(),
+                        recipe.getAuthor(),
+                        recipe.getOrigin(),
+                        recipe.getType().stream().map(RecipeType::getNormalType).toList(),
+                        recipe.getPreparationTime(),
+                        recipe.getTotalTime(),
+                        recipe.getCategory().stream().map(RecipeCategory::getNormalCategory).toList(),
+                        recipe.getDifficulty().getNormalDifficulty(),
+                        recipe.getIngredients()
+                )).findFirst()
+                .orElseThrow(() -> new RecipeNotFoundException(message.replace("${id}", id)));
```

Comment:
Why are you using "RecipeNormalized" and "Recipe" as types/class with the same attributes? 
The repo.findById()-method returns an object Restaurant, which you dont have to parse into a new type RestaurantNormalized. Maybe you can consider to use just one type?